docs: include license checking page (#860)
VeryGoodOpenSource/very_good_cli: site/docs/commands/check_licenses.md

:::

## Arguments ⚙️

### `allowed`

Only allows the use of certain licenses. The command will exit with an error and log the list of all the dependencies that have an unlisted license.

#### Example usage:

```sh
very_good packages check licenses --allowed=MIT,BSD-3-Clause

# ✓ Retrieved 6 licenses from 6 packages of type: BSD-3-Clause (3), MIT (1), unknown (1) and Apache-2.0 (1).
# 2 dependencies have banned licenses: html (unknown) and universal_io (Apache-2.0).
```

:::info
A comprehensive list of all the licenses allowed as options is available within the [_Supported licenses_](#supported-licenses-💳) section of this document.
:::

### `forbidden`

Deny the use of certain licenses. The command will exit with an error and log the list of all the dependencies that have a blocked license.

#### Example usage:

```sh
very_good packages check licenses --forbidden=unknown,Apache-2.0

# ✓ Retrieved 6 licenses from 6 packages of type: BSD-3-Clause (3), MIT (1), unknown (1) and Apache-2.0 (1).
# 2 dependencies have banned licenses: html (unknown) and universal_io (Apache-2.0).
```

:::warning

> The `allowed` and `forbidden` options can't be used at the same time. Typical organization usage dictates which licenses are allowed or forbidden, hence optimizing for that use case.

Author: ```suggestion
The `allowed` and `forbidden` options can't be used at the same time. Typical organization usage dictates which licenses are allowed or forbidden, not both.
```

What do you think of this variant, @tomarra ?

:::

### `dependency-type`

The type of dependencies to check licenses for. There are three available types:

- [`direct-dev`](https://dart.dev/tools/pub/dependencies#dev-dependencies): Another package that your package needs during development.
- [`direct-main`](https://dart.dev/tools/pub/dependencies): Another package that your package needs to work.
- [`transitive`](https://dart.dev/tools/pub/glossary#transitive-dependency): A dependency that your package indirectly uses because one of its dependencies requires it.

When unspecified, it defaults to `direct-main`.

#### Example usage:

```sh
very_good packages check licenses --dependency-type=direct-main,transitive

# ✓ Retrieved 83 licenses from 82 packages of type: BSD-3-Clause (65), MIT (15), unknown (1), BSD-2-Clause (1) and Apache-2.0 (1).
```

:::info
The license checker only requires a [lockfile](https://dart.dev/tools/pub/glossary#lockfile) to gather dependencies. The lockfile is generated automatically for you by [pub][pub] when you run `pub get`, `pub upgrade`, or `pub downgrade`.
:::

### `skip-packages`

> Skips packages from having there licenses checked. Skipped packages will not be checked against `allowed` or `forbidden` licenses.

Reviewer: ```suggestion
Skips packages from having their licenses checked. Skipped packages will not be checked against `allowed` or `forbidden` licenses.
```
